### PR TITLE
[ONNX] Support gather export with OneHot + Mul

### PR DESCRIPTION
**Pull request by BowenBao**
This could serve as a alternative solution to export ```torch.gather``` before something similar goes into ONNX spec. The exported model is verified to be correct against onnxruntime backend. We weren't able to test against Caffe2 backend because it doesn't seem to support OneHot opset9.

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1594,3 +1594,15 @@ def log2(g, self):
 
 def prim_shape(g, self):
     return g.op('Shape', self)
+
+
+@parse_args('v', 'i', 'v', 'v')
+def gather(g, self, dim, index, sparse_grad=False):
+    # NOTE: Update this workaround if ONNX has native Gather support.
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py]: Maybe rephrase to say that current onnx::Gather does not directly support torch.Gather.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1594,3 +1594,15 @@ def log2(g, self):
 
 def prim_shape(g, self):
     return g.op('Shape', self)
+
+
+@parse_args('v', 'i', 'v', 'v')
+def gather(g, self, dim, index, sparse_grad=False):
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py]: This should also support Opset 10 export, correct?

**Review by spandantiwari — APPROVED** [review]

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (houseroad) [reply torch/onnx/symbolic_opset9.py]: Yes, I think so. Should be able to fall back to opset 9's symoblics.
